src/client/pmix_client_pub.c

```diff
@@ -538,11 +538,12 @@ static void wait_cbfunc(struct pmix_peer_t *pr,
                        buf, &ret, &cnt, PMIX_STATUS);
     if (PMIX_SUCCESS != rc) {
         PMIX_ERROR_LOG(rc);
+        ret = rc;
     }
 
   report:
     if (NULL != cb->cbfunc.opfn) {
-        cb->cbfunc.opfn(rc, cb->cbdata);
+        cb->cbfunc.opfn(ret, cb->cbdata);
```

Comment:
See comment on PR #1433 - the earlier error cases are no longer wired to the error mechanism correctly.